### ISSUE TITLE
Standardize the testing for CCCC interface files

### ISSUE THREAD

**Issue by jakehader** [terrapower/armi#287]
Currently, the CCCC interface file testing uses binary files that have been generated from a specific code (e.g., DIF3D). These binary files are used as the basis for reading in the data and checking that it is correct. This exceeds the minimum testing requirement to just ensure that the CCCC interface and read/write ascii and binary files from a given data structure.

It is recommended that we remove the binary files and replace the tests with a lightweight versions. @ntouran and I spoke and think that it would be good to have a `setUpClass` that first creates the Python data structure and then subsequent test methods that will write out the file contents to ascii and binary formats and then read them back in to test that the data is preserved (round trip). There are some nuances to this, especially for Fortran codes that expect to read the binary files with sequential access so we should test that functionality as well. This is represented by the stream's `createRecord` method.

This will improve the portability of the CCCC interface file tests and decouple them from the specific way in which they were originally generated (i.e., from DIF3D, PARTISN, etc.).

This is related to https://github.com/terrapower/armi/pull/286#pullrequestreview-644767347.

**Comment by ntouran** [comment]
Yeah, minimizing testing dependencies on non-portable binary blobs of mystery is less than ideal.

**Comment by Scienkiller** [comment]
Hello, I am very glad to meet you, and ARMI is a very interesting thing. These days of study has given me a general understanding of the functions and usage of ARMI, but there are still many unclear points about the software coupling. If you can, I hope you can teach me, or give me some tips, and I will thank you very much

**Comment by ntouran** [comment]
Hi @Scienkiller thanks for the question. We are trying to do Q&A  in the Github discussions feature. Would you mind re-posting that question here? https://github.com/terrapower/armi/discussions/categories/q-a

**Comment by Scienkiller** [comment]
Okay, thank you very much for your reply, I am very happy to do this.

**Comment by john-science** [comment]
This ticket was closed with the PR: https://github.com/terrapower/armi/pull/290
